packages/client-sdk-nodejs/src/internal/cache-data-client.ts

```diff
@@ -386,15 +386,17 @@ export class CacheDataClient implements IDataClient {
     }
 
     try {
-      await this.requestConcurrencySemaphore.acquire();
+      if (this.requestConcurrencySemaphore !== undefined)
```

Comment:
i feel like we should be able to dry this up, e.g. by creating a generic function higher-order called `rateLimited(() => T): T` and then doing `return await this.rateLimited(() => this.sendSet(...))`.

Not a blocker for this PR as the most important thing is to fix the behavior, which I think you've done here. Maybe in a follow-up PR though.